decidim-core/app/models/decidim/component.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+module Decidim
+  # A Component represents the association between a participatory process and a
+  # Component Engine. It configures how that component should behave, its name,
+  # and other relevant fields.
+  class Component < ApplicationRecord
+    belongs_to :feature, foreign_key: "decidim_feature_id"
+
+    belongs_to :step,
+               foreign_key: "decidim_participatory_process_step_id",
+               class_name: "Decidim::ParticipatoryProcessStep"
+
+    has_one :participatory_process, through: :feature
+
+    validates :participatory_process, :step, presence: true
```

Comment:
Not sure but: shouldn't the feature be mandatory? If it's a has_one relation, why are we validating the presence of participatory_process? (Maybe you want to add a validate: true to the belongs_to :step to make sure it's valid, or use validates_associated)